### PR TITLE
Logging system rework

### PR DESCRIPTION
**Pull request by MRiganSUSX**
Please see details in https://github.com/DUNE-DAQ/trigger/pull/285

### CODE REVIEW

File: include/triggeralgs/Logging.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+enum
+{
+  TLVL_VERY_IMPORTANT = 1,
+  TLVL_IMPORTANT      = 2,
+  TLVL_GENERAL        = 3,
+  TLVL_DEBUG_INFO     = 4,
+  TLVL_DEBUG_LOW      = 5,
+  TLVL_DEBUG_MEDIUM   = 10,
+  TLVL_DEBUG_HIGH     = 15,
+  TLVL_DEBUG_ALL      = 20
```
Comment (ArturSztuc) [inline include/triggeralgs/Logging.hpp:27]: `IMPORTANT` and `VERY_IMPORTANT` is strange naming (I guess `GENERAL` and `DEBUG_INFO` are too). Is the naming similar across the rest of the DAQ? The usual naming for different logging verbosity in the industry is:
* `TRACE` for everything, so `TLVL_DEBUG_ALL`
* `DEBUG` for the debugging messages (here it makes sense to have low, medium, high)
* `INFO` would be your `VERY_IMPORTANT`, `IMPORTANT` and/or `DEBUG_INFO`,
* `WARN` for warnings (we have different system for that),
* `ERROR` for errors (we have different system for that),
* `FATAL` for fatal errors that should crash (we have different system for that).

What are the namings in the rest of the DAQ?

File: src/TriggerCandidateMakerMichelElectron.cpp [hunk 2]
```diff
@@ -34,7 +40,7 @@ TriggerCandidateMakerMichelElectron::operator()(const TriggerActivity& activity,
 
       // add_window_to_record(m_current_window);
       // dump_window_record();
-      // TLOG(1) << "Constructing trivial TC.";
+      TLOG_DEBUG(TLVL_DEBUG_LOW) << "[TCM:ME] Constructing trivial TC.";
```
Comment (ArturSztuc) [inline src/TriggerCandidateMakerMichelElectron.cpp]:
```suggestion
      TLOG_DEBUG(TLVL_DEBUG_LOW) << "[TCM:ME] Constructing a TC.";
```

I know this is not your doing, but might as well correct this... why "Constructing trivial TC"? Why "trivial"? Are there non-trivial TCs? I think just "Constructing a TC" would be fine here, and less confusing...

File: src/TriggerCandidateMakerPlaneCoincidence.cpp [hunk 3]
```diff
@@ -34,8 +40,8 @@ TriggerCandidateMakerPlaneCoincidence::operator()(const TriggerActivity& activit
 
       // add_window_to_record(m_current_window);
       // dump_window_record();
-      TLOG(1) << "Constructing trivial TC.";
-      TLOG(1) << "Activity count: " << m_activity_count;
+      TLOG_DEBUG(TLVL_DEBUG_LOW) << "[TCM:PC] Constructing trivial TC.";
```
Comment (ArturSztuc) [inline src/TriggerCandidateMakerPlaneCoincidence.cpp]:
```suggestion
      TLOG_DEBUG(TLVL_DEBUG_LOW) << "[TCM:PC] Constructing a TC.";
```

Ditto w.g. "trivial"

**Review by ArturSztuc — CHANGES_REQUESTED** [review]
Overall looks good to me, although I find the trigger logging level names strange

File: include/triggeralgs/Logging.hpp (hunk 1 above)
Comment (MRiganSUSX) [reply include/triggeralgs/Logging.hpp:27]: This naming is completely custom. As far as I know, there is no consistency across dunedaq. There are some repos that have a version of this, example [in-file enum here](https://github.com/DUNE-DAQ/flxlibs/blob/production/v4/src/CardWrapper.cpp#L26-L31) and [separare logging header here](https://github.com/DUNE-DAQ/readoutlibs/blob/production/v4/include/readoutlibs/ReadoutLogging.hpp#L18-L29). Other places simply use integers [example here](https://github.com/DUNE-DAQ/hsilibs/blob/production/v4/plugins/HSIReadout.cpp#L250), and these do go to high values ([example here](https://github.com/DUNE-DAQ/iomanager/blob/production/v4/src/network/ConfigClient.cpp#L89)).

I just made that list according to what made sense to me, and wanted it to (a) be granural at the lower levels and (b) roughly cover the available space for levels. Open to changes. As you mentioned, we are not using standards from the industry (or at least not closely).

File: src/TriggerCandidateMakerMichelElectron.cpp (hunk 2 above)
Comment (MRiganSUSX) [reply src/TriggerCandidateMakerMichelElectron.cpp]: I have not been generally modifying the messages (outside of few examples in MLT where I added more information). I agree with your change.

**Comment by MRiganSUSX** [comment]
Thanks for the review. I have removed the 'trivial' messages in comment above.

**Review by ArturSztuc — APPROVED** [review]

**Pull request merged**